components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -28,10 +28,30 @@ import (
 )
 
 // compile time check
-var _ MessagingBackend = &Nats{}
+var _ NatsBackend = &Nats{}
+
+const (
+	backoffStrategy = cev2context.BackoffStrategyConstant
+	natsHandlerName = "nats-handler"
+)
 
 type ConnClosedHandler func(conn *nats.Conn)
 
+type NatsBackend interface {
+	// Initialize connects and initializes the NATS backend.
+	// connCloseHandler can be used to register a handler that gets called when connection
+	// to the NATS server is closed and retry attempts are exceeded.
+	Initialize(connCloseHandler ConnClosedHandler) error
+
+	// SyncSubscription synchronizes the Kyma Subscription on the NATS backend.
+	// It returns true if Kyma Subscription status was changed during this synchronization process.
+	// It sets subscription.status.config with configurations that were applied on the messaging backend when creating the subscription.
+	SyncSubscription(subscription *eventingv1alpha1.Subscription, params ...interface{}) (bool, error)
```

Comment:
I'll try to get rid of the variadics.

Reply:
https://github.com/kyma-project/kyma/pull/13494/commits/f5977207884b0564674088686f110d64fd6ff8ea